hyrisecockpit/database_manager/database.py

```diff
@@ -109,21 +112,39 @@ def get_queue_length(self) -> int:
         """Return queue length."""
         return self._worker_pool.get_queue_length()
 
-    def load_data(self, folder_name: str) -> bool:
+    def load_data(self, workload: Dict) -> bool:
         """Load pre-generated tables."""
-        return (
-            False
-            if self._worker_pool.get_status() != "closed"
-            else self._background_scheduler.load_tables(folder_name)
-        )
+        workload_type = workload["workload_type"]
+        scale_factor = workload["scale_factor"]
+        if workload_type not in list(self._workload_drivers.keys()):
+            return False
+        elif (
+            scale_factor not in self._workload_drivers[workload_type].get_scalefactors()
+        ):
+            return False
+        elif self._worker_pool.get_status() != "closed":
+            return False
+        else:
+            return self._background_scheduler.load_tables(
+                workload_type, float(scale_factor)
+            )
 
-    def delete_data(self, folder_name: str) -> bool:
+    def delete_data(self, workload: Dict) -> bool:
         """Delete tables."""
-        return (
-            False
-            if self._worker_pool.get_status() != "closed"
-            else self._background_scheduler.delete_tables(folder_name)
-        )
+        workload_type = workload["workload_type"]
+        scale_factor = workload["scale_factor"]
+        if workload_type not in list(self._workload_drivers.keys()):
```

Comment:
```suggestion
        if workload_type not in self._workload_drivers:
```
https://stackoverflow.com/a/1602944